server/src/main/java/io/crate/planner/operators/InsertFromValues.java

```diff
@@ -453,10 +465,16 @@ public CompletableFuture<BulkResponse> executeBulk(DependencyCarrier dependencie
         }).whenComplete((response, t) -> {
             if (t == null) {
                 bulkResponse.update(response, bulkIndices);
-                result.complete(bulkResponse);
             } else {
-                result.completeExceptionally(t);
+                assert response == null : "No response when one bulk short circuited." +
+                    "Creating a bulkResponse by setting up the same error to all bulks.";
+                for (IntCursor c : bulkIndices) {
+                    int resultIdx = c.value;
+                    t = SQLExceptions.prepareForClientTransmission(accessControl, t);
+                    bulkResponse.setFailure(resultIdx, t);
+                }
             }
+            result.complete(bulkResponse);
```

Comment:
I don't understand why we're changing the bulk response handling too?

It already has bulk parameter dependend `Throwable[] failures` - all we need to do is to ensure the throwable/failures are set, then the reporting to the client (and here) should all already fit. Or what am I missing?

Reply:
It all boils down to exceptions being swallowed on emitting final result, ie we don't have 
`exceptionally` in addition to
` .thenAccept(bulkResp -> emitRowCountsToResultReceivers(jobId, jobsLogs, toExec, bulkResp));`

The thing is  `result.completeExceptionally(t);` was never hit on master
(because we never had `continueOnError` flipped)

If I revert this, `test_bulk_insert_from_values_fail_fast` gets stuck. Then it kinda works if I add `exceptionally` to the Session (to catch bubbled up exception) but there we have only error, no `BulkResponse`, so we don't know which ResultResiver we should fail and which one just finish.

Reply:
bulkresponse.failures contains 2 nulls BTW, we need to enrich it by the actual failure we caught.

When future completes with a proper response, we can build response properly via  `bulkResponse.update(response, bulkIndices);` when we have only error, we build response based on the failure